elements/rh-icon/rh-icon.ts

```diff
@@ -0,0 +1,174 @@
+ * dimensions in the same area without shifting surrounding content.
+ * @summary Decorative element which supports related content
+ * @slot - Slotted content is used as a fallback in case the icon doesn't load
+ * @fires load - Fired when an icon is loaded and rendered
+ * @fires error - Fired when an icon fails to load
+ * @csspart fallback - Container for the fallback (i.e. slotted) content
+ * @cssprop --rh-icon-size - Override default icon size
+ */
+@customElement('rh-icon')
+export class RhIcon extends LitElement {
+  public static readonly styles: CSSStyleSheet[] = [style];
+
+  private static onIntersect: IntersectionObserverCallback = records =>
+    records.forEach(({ isIntersecting, target }) => {
+      const icon = target as RhIcon;
+      icon.#intersecting = isIntersecting;
+      ric(() => {
+        if (icon.#intersecting) {
+          icon.load();
+        }
+      });
+    });
+
+
+  private static io = new IntersectionObserver(RhIcon.onIntersect);
+
+  private static instances = new Set<RhIcon>();
+
+  private static resolve: IconResolverFunction = (set: string, icon: string): Renderable =>
+    import(`@rhds/icons/dist/${set}/${icon}.js`)
+        .then(mod => mod.default.cloneNode(true));
+
+  /** Icon set */
+  @property() set = 'standard';
+
+  /** Icon name */
+  @property({ reflect: true }) icon = '';
+
+  /**
+   * Defines a string value that labels the icon element.
+   * Adds role="img" to element.
+   */
+  @property({ attribute: 'accessible-label' }) accessibleLabel?: string;
+
+  /**
+   * Controls how eager the element will be to load the icon data
+   * - `eager`: eagerly load the icon, blocking the main thread
+   * - `idle`: wait for the browser to attain an idle state before loading
+   * - `lazy` (default): wait for the element to enter the viewport before loading
+   */
+  @property() loading?: 'idle' | 'lazy' | 'eager' = 'lazy';
+
+  /** Icon content. Any value that lit can render */
+  @state() private content?: unknown;
+
+  #intersecting = false;
+
+  #logger = new Logger(this);
+
+  #internals = InternalsController.of(this);
+
+  #lazyLoad() {
+    RhIcon.io.observe(this);
+    if (this.#intersecting) {
+      this.load();
+    }
+  }
+
+  #load() {
+    switch (this.loading) {
+      case 'idle': return void ric(() => this.load());
+      case 'lazy': return void this.#lazyLoad();
+      case 'eager': return void this.load();
+    }
+  }
+
+  async #contentChanged() {
+    await this.updateComplete;
+    this.dispatchEvent(new Event('load', { bubbles: true }));
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    RhIcon.instances.add(this);
+    if (this.accessibleLabel) {
+      this.#internals.ariaLabel = this.accessibleLabel;
+      this.#internals.role = 'img';
+    }
+  }
+
+  willUpdate(changed: PropertyValues<this>): void {
+    if (changed.has('icon')) {
+      this.#load();
+    }
+  }
+
```

Comment:
```suggestion
  connectedCallback(): void {
    super.connectedCallback();
    RhIcon.instances.add(this);
  }

  @observes('icon')
  private iconChanged(): void {
    this.#load();
  }
  
  @observes('accessibleLabel')
  private accessibleLabelChanged(): void {
    this.#internals.ariaLabel = this.accessibleLabel;
    if (this.accessibleLabel) {
      this.#internals.role = 'img';
    } else {
      this.#internals.role = 'presentation';
    }
  }

```
please confirm @adamjohnson

Reply:
this will wait until pfe 4 is out and installed in this repo

Reply:
Works for me. 

If you wanted to be pedantic, we could not set a role for the `else`. That said, it's probably six in one, half a dozen in the other.

Reply:
is the current iteration good?